tests/e2e/__testers__.ts

```diff
@@ -4,29 +4,68 @@ import { values } from 'lodash'
 import stripAnsi from 'strip-ansi'
 import { ClientBase } from '@prisma-spectrum/reflector/dist-cjs/Client'
 import { Reflector } from '@prisma-spectrum/reflector'
+import { getPrismaClient } from '~/tests/e2e/helpers/getPostgresAdminPrismaClient'
+import { PrismaClient } from '@prisma/client'
 
-export const testTemplate = (params: {
+export interface DBTestParams {
   templateName: PrismaTemplates.$Types.Template['_tag']
-  /** A string or regex pattern of the expected output (stdout) when the project dev mode is run.*/
   expectedDevOutput: RegExp | string
-}) => {
-  jest.setTimeout(50_000)
+  datasourceProvider: Reflector.Schema.DatasourceProviderNormalized
+  connectionStringBase: string
+  getPrismaAdmin: getPrismaClient
+  prismaConfig?: {
+    referentialIntegrity?: 'prisma' | 'foreignKeys'
+  }
+}
+
+async function dropDatabase(prismaClient: PrismaClient, databaseName: string, datasourceProvider: Reflector.Schema.DatasourceProviderNormalized) {
+  switch (datasourceProvider){
+    case 'postgres':
+      try {
+        return await prismaClient.$executeRawUnsafe(`DROP DATABASE ${databaseName} WITH (FORCE);`)
+      } catch (error) {
+        const isDatabaseNotFoundErorr = error instanceof Error && error.message.match(/does not exist/)
+        if (!isDatabaseNotFoundErorr) throw error
+        return;
+      }
+    case 'mysql':
+      try {
+        return await prismaClient.$executeRawUnsafe(`DROP DATABASE IF EXISTS ${databaseName};`)
+      } catch (error) {
+        const isDatabaseNotFoundErorr = error instanceof Error && error.message.match(/database not found/)
+        if (!isDatabaseNotFoundErorr) throw error
+        return;
+      }
+  }
+}
+
+async function initDatabase(prismaClient: PrismaClient, databaseName: string) {
+  return await prismaClient.$executeRawUnsafe(`CREATE DATABASE IF NOT EXISTS ${databaseName}`)
+}
```

Comment:
Instead of wrapping `prismaClient.$executeRawUnsafe` could we just work with data in that we need only switch the instructions that are passed to `executeRawUnsafe`?  I could see this wrapper becoming useful for Mongodb if its significantly different for this part, though.

Reply:
Not sure what needs to be done here. Could you show with an example, please?

Reply:
Not a big deal right now but in tests we can do:

```ts
await prismaClient.$executeRawUnsafe(getCreateDatabaseInstruction(params.datasourceProvider))
```

More direct. `Instruction` rather than `SQL` because of forthcoming `Mongo` etc. (but as I said I'm not sure what will happen when we get to Mongo, maybe it will require a complete function wrapping like this to keep the abstraction encapsulated...)

Reply:
Discussed on slack. Basically this is outdated, and there is a switch case now, for Postgres and MySQL.